guides/token.md

```diff
@@ -68,3 +68,33 @@ import liboidcagent as agent
 
 token, issuer, expires_at = agent.get_token_response("apex_project_a")
 ```
+
+
+## Machine to machine login via client credentials
+
+While the interactive login is the most secure option, sometimes you require a long running script script or service to 
+interact with an instantiation service like the catalog. In this case, you can use the client credentials flow, which is
+basically a username/password login for machines. 
+
+It is recommended to create such client credentials for each service separately, with minimal privileges. This allows you
+to retain granular access control, and to revoke access to a specific service without affecting others.
+
+To obtain the client credentials, you currently need to contact the APEx support team.
+
+The Python script below demonstrates how to obtain a token using the client credentials flow, which comes down to a very
+simply http request that can also be performed easily via curl or other tools.
+
+```python
+import requests
+scopes = ["openid", "roles" ]
+token_response = requests.post(
+    "https://auth.apex.esa.int/realms/apex/protocol/openid-connect/token",
+    data={
+        "grant_type": "client_credentials",
+        "client_id": "weed_catalogue_api",
+        "client_secret": "XXX",
```

Comment:
This snippet shows a hardcoded `weed_catalogue_api` client id. I think it should be more clear that client id is variable with a value that is custom for the user.
The client secret part also promotes hardcoding secrets in source code.

A low tech solution to tackle both problems in this simple snippet is getting the values from env vars, e.g.

```python
        "client_id": os.environ["APEX_CLIENT_ID"],
        "client_secret": os.environ["APEX_CLIENT_SECRET"],
```

It probably makes correctly running this script a bit more annoying, but at least it's more secure out of the box in the documentation (and it's not hard for users to figure out to do it easier, at the cost of being less secure, but that's their responsibility then)